File: backend/internal/models/property.go
package models

import "time"

type Property struct {
	ID            int       `json:"id"`
	Name          string    `json:"name"`
	Address       string    `json:"address"`
	PropertyType  string    `json:"property_type"`
	DefaultMileage float64  `json:"default_mileage"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type CreatePropertyRequest struct {
	Name          string  `json:"name"`
	Address       string  `json:"address"`
	PropertyType  string  `json:"property_type"`
	DefaultMileage float64 `json:"default_mileage"`
}

File: frontend/src/types/property.ts
export interface Property {
  id: number
  name: string
  address: string
  property_type: string
  default_mileage: number
  created_at: string
  updated_at: string
}

export interface CreatePropertyRequest {
  name: string
  address: string
  property_type: string
  default_mileage?: number
}
